src/utils/handleRequestWith.ts

```diff
@@ -92,6 +103,9 @@ export const handleRequestWith = (
         return channel.send({ type: 'MOCK_NOT_FOUND' })
       }
 
+      store.add(<Request>(<unknown>req), <Response>(<unknown>response))
+      store.persist()
```

Comment:
With the cookie store persisting on each request handling, we need to seriously consider failure scenarios. 

- Local storage entry may become corrupted.
- Failures to sync cookies with the storage must not raise an exception and fail a request.

Perhaps we could hydrate/persist at a different step? What advantages to you see in doing so upon each request handling?

Reply:
I suppose a corrupted persisted value is handled as a part of https://github.com/mswjs/cookies/pull/2.